patchwork-extensions-shearing/src/main/java/com/patchworkmc/mixin/extensions/shearing/MixinShearsItem.java

```diff
@@ -0,0 +1,56 @@
+package com.patchworkmc.mixin.extensions.shearing;
+
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.enchantment.Enchantments;
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ShearsItem;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.IShearable;
+import org.spongepowered.asm.mixin.Implements;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Patch ShearsItem to allow it to shear any IShearable.
+ *
+ * @author SuperCoder79
+ */
+@Mixin(ShearsItem.class)
+public class MixinShearsItem extends Item {
+
+	public MixinShearsItem(Settings settings) {
+		super(settings);
+	}
+
+	@Override
+	public boolean useOnEntity(ItemStack stack, PlayerEntity playerIn, LivingEntity entity, Hand hand) {
+		if (entity.world.isClient) return false;
+
+		if (entity instanceof IShearable) {
```

Comment:
Lines 39 to 48 are identical to lines 51 to line 59 in `MixinDispenserBehavior`

Reply:
Perhaps we could make a `Shearables.shearEntity` function to avoid code duplication here.